xbmc/guilib/Interpolators.h

```diff
@@ -0,0 +1,363 @@
+/*
+ *  Copyright (C) 2005-2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+///////////////////////////////////////////////////////////////////////
+// Interpolators.h
+// A couple of tweening classes implemented in C++.
+// ref: http://www.robertpenner.com/easing/
+//
+// Author: d4rk <d4rk@xbmc.org>
+///////////////////////////////////////////////////////////////////////
+
+#include <math.h>
+
+#ifndef M_PI
+#define M_PI 3.14159265358979323846
+#endif
+
+enum class EASE
+{
+  IN,
```

Comment:
As commented Neo1973 Windows build fails because IN, OUT already defined. Possible fix without change names is add this before enum class:

```c++
#ifdef IN
#undef IN
#endif

#ifdef OUT
#undef OUT
#endif
```

Although I'm not entirely sure if it can cause any unwanted effects...